Client/Frontend/Browser/URLBarView.swift

```diff
@@ -37,28 +37,52 @@ class URLBarView: UIView, BrowserLocationViewDelegate, UITextFieldDelegate {
 
     private var locationView: BrowserLocationView!
     private var editTextField: ToolbarTextField!
-    private var locationContainer: UIView!
-    private var tabsButton: UIButton!
+    var locationContainer: UIView!
+    var tabsButton: UIButton!
     private var progressBar: UIProgressView!
-    private var cancelButton: UIButton!
+    var cancelButton: UIButton!
     private var curveShape: CurveView!
 
+    weak var browserToolbarDelegate: BrowserToolbarDelegate?
+
+    var shareButton: UIButton
+    var bookmarkButton: UIButton
+    var forwardButton: UIButton
+    var backButton: UIButton
+    var helper: BrowserToolbarHelper?
+    var showToolbar = false
+
     override init() {
+        self.shareButton = UIButton()
+        self.bookmarkButton = UIButton()
+        self.forwardButton = UIButton()
+        self.backButton = UIButton()
         // super.init() calls init(frame: CGRect)
         super.init()
     }
 
     override init(frame: CGRect) {
+        self.shareButton = UIButton()
+        self.bookmarkButton = UIButton()
+        self.forwardButton = UIButton()
+        self.backButton = UIButton()
         super.init(frame: frame)
         initViews()
     }
 
     required init(coder aDecoder: NSCoder) {
+        self.shareButton = UIButton()
+        self.bookmarkButton = UIButton()
+        self.forwardButton = UIButton()
+        self.backButton = UIButton()
```

Comment:
Why are you duplicating all of these in each initializer? Can't you initialize all of them at the top level? That is:
```
let shareButton = UIButton()
let bookmarkButton = UIButton()
...
```

Also, can you define all of these buttons with `let` instead of `var`?